inc/Engine/Media/AboveTheFold/Factory.php

```diff
@@ -4,22 +4,34 @@
 
 namespace WP_Rocket\Engine\Media\AboveTheFold;
 
-use WP_Rocket\Engine\Common\PerformanceHints\AbstractFactory;
+use WP_Rocket\Engine\Common\PerformanceHints\FactoryInterface;
+use WP_Rocket\Engine\Common\PerformanceHints\AJAX\ControllerInterface as AjaxControllerInterface;
 
-class Factory implements AbstractFactory {
+class Factory implements FactoryInterface {
+
+	/**
+	 * Ajax Controller.
+	 *
+	 * @var AjaxControllerInterface
+	 */
+	protected $ajax_controller;
 
 	/**
 	 * Instatiate the class.
+	 *
+	 * @param AjaxControllerInterface $ajax_controller ATF AJAX Controller.
 	 */
-	public function __construct() {
-		// Assign objects to their properties.
+	public function __construct( AjaxControllerInterface $ajax_controller ) {
+		$this->ajax_controller = $ajax_controller;
 	}
 
 	/**
-	 * Provides an Ajax object.
+	 * Provides an Ajax controller object.
+	 *
+	 * @return AjaxControllerInterface
 	 */
-	public function ajax() {
-		// Return Ajax object.
+	public function ajax(): AjaxControllerInterface {
```

Comment:
How about renaming this to `get_ajax_controller()` to make it clear what is returned by the method?